### DISCUSSION TITLE
ipynb vs dib  - secrets and SCM reading file

### DISCUSSION THREAD

**Discussion by mluker** [dotnet/interactive#2001] [Q&A] [answered]
When using ipynb format the output is stored within the notebook which may contains secrets, connection strings, etc. When using dib format it looks that's not the case which is better for me, I don't want secrets in source control. The issue with dib is Azure DevOps doesn't display dib files as text. Is there a workaround for this?

**Comment by jonsequitur** [comment, marked as answer]
Currently, no, although we've discussed adding support for a Markdown format like [R Markdown](https://rmarkdown.rstudio.com/) or the [Try .NET Markdown format](https://www.hanselman.com/blog/introducing-the-try-net-global-tool-interactive-inbrowser-documentation-and-workshop-creator).

Another approach that can be used to keep secrets out of your notebook was just released today so there's no documentation for it yet, but you can give it a try if you're using VS Code.

<img width="730" alt="image" src="https://user-images.githubusercontent.com/547415/167505055-bc3b83a9-7b76-4ada-a901-86ee12281ed2.png">

<img width="721" alt="image" src="https://user-images.githubusercontent.com/547415/167505244-b57b628c-a31f-404c-ba97-f96dd38ba8d7.png">

**Reply by mluker** [reply]
Thank you for the response, I appreciate it!

**Reply by jonsequitur** [reply]
You're welcome. Let us know if this works for you and if there's anything you'd improve.

**Reply by cecilphillip** [reply]
@jonsequitur any update on the docs for this?

**Reply by jonsequitur** [reply]
Sorry for the late response. Updated docs are here: https://github.com/dotnet/interactive/blob/main/docs/input-prompts.md#passwords-and-other-secrets
